pkg/core/blockchain.go

```diff
@@ -815,6 +856,15 @@ func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error
 		mpt.Collapse(10)
 	}
 
+	err, ok := <-aerdone
+	if ok && err != nil {
+		return err
+	}
+	err, ok = <-blockdone
+	if ok && err != nil {
+		return err
+	}
```

Comment:
It is not obvious when block goroutine exits if we return after error in `aerdone`. It can hang forever on channel send (as well as perform `Persist` concurrently). Am I missing something?

Reply:
Persist is not a problem, but channel send might be one, true.

Reply:
Fixed.